File: editorial/chap03/answer_B12.py
def f(x):
	return x*x*x + x

N = int(input())

# 二分探索
Left = 0.0
Right = 100.0
for i in range(20):
	Mid = (Left + Right) / 2
	val = f(Mid)

	# 探索範囲を絞る
	if val > N:
		Right = Mid # 左半分に絞られる
	else:
		Left = Mid # 右半分に絞られる

# 出力
print(Mid)

File: editorial/chap03/cpp/answer_B12.cpp
#include <iostream>
using namespace std;
 
// 関数 f
double f(double x) {
	return x * x * x + x;
}
 
int main() {
	// 入力
	int N;
	cin >> N;
 
	// 二分探索
	double Left = 0, Right = 100, Mid;
	for (int i = 0; i < 20; i++) {
		Mid = (Left + Right) / 2.0;
		double val = f(Mid);
 
		// 探索範囲を絞る
		if (val > 1.0 * N) Right = Mid; // 左半分に絞られる
		else Left = Mid; // 右半分に絞られる
	}
 
	// 出力
	printf("%.12lf\n", Mid);
	return 0;
}
